lib/srv/sshserver_test.go

```diff
@@ -238,7 +239,8 @@ func (s *SrvSuite) TestAgentForward(c *C) {
 	_, err = io.WriteString(writer, fmt.Sprintf("printenv %v\n\r", teleport.SSHAuthSock))
 	c.Assert(err, IsNil)
 
-	re := regexp.MustCompile(`/tmp/[^\s]+`)
+	pattern := fmt.Sprintf(`%vteleport-[0-9]+[^\s]+`, os.TempDir())
+	re := regexp.MustCompile(pattern)
```

Comment:
This is not actually the line that has the problem, but it's the loop below:

```go
 	for i := 0; i < 3; i++ {
 		_, err = reader.Read(buf)
 		c.Assert(err, IsNil)
 		matches = re.FindStringSubmatch(string(buf))
 		if len(matches) != 0 {
 			break
 		}
 		time.Sleep(50 * time.Millisecond)
 	}
```

On Linux tests hang on the line `_, err = reader.Read(buf)`. Works fine on macOS.